src/decoder_wav.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "system.h"
+
+#ifdef WANT_FASTWAV
+
+// Headers
+#include <cassert>
+#include <sys/stat.h>
+#include "decoder_wav.h"
+#include "output.h"
+
+WavDecoder::WavDecoder() 
+{
+	music_type = "wav";
+}
+
+WavDecoder::~WavDecoder() {
+	if (file_ != NULL) {
+		fclose(file_);
+	}
+}
+
+bool WavDecoder::Open(FILE* file) {
+	file_=file;
+	uint32_t chunk = 0xDEADBEEF;
+	uint32_t jump;
+	fseek(file_, 22, SEEK_SET);
+	fread(&nchannels, 1, 2, file_);
+	fread(&samplerate, 1, 4, file_);
+	uint16_t bitspersample;
+	fread(&bitspersample, 1, 2, file_);
+	switch (bitspersample) {
```

Comment:
One thing to note here is that there are no error checks for invalid wave files (i.e. bad sample rate). But given the games have mostly been tested before, this problem should be minimal.